index_websocket.js

```diff
@@ -0,0 +1,97 @@
+const http = require("http");
+const open = require("open");
+const fs = require("fs");
+const WebSocketServer = require("websocket").server; // websocket server class
+
+let webSocket = null;
+// connection variable, to refer to the web socket connection to the client once connection is established
+let connection = null;
+// default port used if none is specified by the user
+let defaultPort = 9101;
+const host = "http://127.0.0.1";
+
+// check if a browser window is already open
+let isRunning = false;
+
+
+exports.visualise = function visualise(jsonData, port = defaultPort) {
+
+  // overhead of server creation only done on first call
+  if (!isRunning) {
+    isRunning = true;
+
+    // open index.html in the browser
+    open("http://127.0.0.1:" + port);
```

Comment:
Let's open the browser at the end.

Reply:
Reason for opening it at the start was to establish the connection as soon as possible